File: 2022/25/rust/src/main.rs
use rust::snafu;
use std::fs::File;
use std::io::{ BufRead, BufReader };

// --- Constants ---
const INPUT_FILE: &'static str = "../input/input.txt";

// --- Main function ---
fn main() -> std::io::Result<()> {
    let file = File::open(INPUT_FILE)?;
    let total = BufReader::new(file).lines().map(|s| snafu::from(&s.unwrap())).sum();
    let snafu = snafu::to(total);
    println!("SNAFU number to supply to Bob's console: {}", snafu);
    Ok(())
}

File: 2022/25/python/main.py
import snafu

# --- Constants ---
INPUT_FILE = "../input/input.txt"

# --- Read and parse the input file ---
with open(INPUT_FILE, "r") as file:
    text = file.read()

total = sum(map(lambda line: snafu.from_snafu(line), text.strip().split("\n")))
snafu = snafu.to_snafu(total)
print(f"SNAFU number to supply to Bob's console: {snafu}")
